include/boost/geometry/algorithms/hausdorff_distance.hpp

```diff
@@ -0,0 +1,324 @@
+// Boost.Geometry
+
+// Copyright (c) 2018 Yaghyavardhan Singh Khangarot, Hyderabad, India.
+
+// Contributed and/or modified by Yaghyavardhan Singh Khangarot, as part of Google Summer of Code 2018 program.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#include <algorithm> 
+
+#ifdef BOOST_GEOMETRY_DEBUG_HAUSDORFF_DISTANCE
+#include <iostream>
+#endif
+
+#include <iterator>
+#include <utility>
+#include <vector>
+#include <limits>
+
+#include <boost/geometry/geometry.hpp>
+#include <boost/geometry/geometries/linestring.hpp>
+#include <boost/geometry/geometries/point_xy.hpp>
+#include <boost/geometry/geometries/polygon.hpp>
+#include <boost/range.hpp>
+#include <boost/mpl/assert.hpp>
+#include <boost/geometry/core/tag.hpp>
+#include <boost/geometry/index/rtree.hpp>
+
+namespace bgi = boost::geometry::index;
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace hausdorff_distance
+{
+
+struct point_range
+{
+    template <typename Point, typename Range, typename Strategy>
+    static inline typename distance_result
+        <
+            typename point_type<Point>::type,
+            typename point_type<Range>::type,
+            Strategy
+        >::type apply(Point const& pnt, Range const& rng, Strategy const& strategy)
+    {
+        
+        typedef typename distance_result
+            <
+                typename point_type<Point>::type,
+                typename point_type<Range>::type,
+                Strategy
+            >::type result_type;
+        typedef typename boost::range_size<Range>:: type size_type;
+        
+        boost::geometry::detail::throw_on_empty_input(pnt);
+        boost::geometry::detail::throw_on_empty_input(rng);
```

Comment:
Right, so if this function is called below in `range_range` this check is performed in each iteration of the higher-level loop. It would be good to avoid it.

Reply:
Removed in the updated code.